t/config.t

```diff
@@ -3,59 +3,76 @@ use Test2::V0;
 use File::Share 'dist_file';
 use NewFangle::Agent::Config;
 
-$ENV{NEWRELIC_CONFIG_FILE} = dist_file 'NewFangle-Agent', 'test.yml';
-NewFangle::Agent::Config::initialize;
-
-like +NewFangle::Agent::Config->global_settings => {
-    enabled     => T,
-    license_key => 'DEADBEEF',
-    app_name    => 'Perl Application',
-    log_level   => 'error',
-    transaction_tracer => {
-        enabled => 1,
-        include => {
-            paths => [
-                't/lib',
-            ],
-        },
-    }
-} => 'Loaded test config';
-
-$ENV{NEWRELIC_ENVIRONMENT} = 'environment';
-
-NewFangle::Agent::Config->initialize; # Can call as class method
-
-my $global = NewFangle::Agent::Config->global_settings;
-
-is $global->{license_key}, 'eu01xxdeadbeefdeadbeefdeadbeefdeadbeNRAL',
-    'Environment overwrites config';
-
-is +NewFangle::Agent::Config->local_settings, $global,
-    'Unchanged local settings same as global';
-
-$global->{app_name} = 'Fake name';
-is +NewFangle::Agent::Config->local_settings->{app_name}, 'Perl Application',
-    'Config is read-only';
-
-$global = NewFangle::Agent::Config->global_settings;
-{
-    local $ENV{NEWRELIC_ENABLED} = $global->{enabled} ? 1 : 0;
+subtest 'Config file from environment' => sub {
+    local $ENV{NEWRELIC_CONFIG_FILE} = dist_file 'NewFangle-Agent', 'test.yml';
+
+    NewFangle::Agent::Config::initialize;
+
+    like +NewFangle::Agent::Config->global_settings => {
+        enabled     => T,
+        license_key => 'DEADBEEF',
+        app_name    => 'Perl Application',
+        log_level   => 'error',
+        transaction_tracer => {
+            enabled => 1,
+            include => {
+                paths => [
+                    't/lib',
+                ],
+            },
+        }
+    };
+};
+
+subtest 'New Relic environment override' => sub {
+    local $ENV{NEWRELIC_CONFIG_FILE} = dist_file 'NewFangle-Agent', 'test.yml';
+    local $ENV{NEWRELIC_ENVIRONMENT} = 'environment';
+
+    NewFangle::Agent::Config->initialize; # Can call as class method
+
+    my $global = NewFangle::Agent::Config->global_settings;
+
+    is $global->{license_key}, 'eu01xxdeadbeefdeadbeefdeadbeefdeadbeNRAL',
+        'Environment overwrites config';
+
+    is +NewFangle::Agent::Config->local_settings, $global,
+        'Unchanged local settings same as global';
+
+    is +NewFangle::Agent::Config->local_settings->{app_name}, 'Perl Application',
```

Comment:
Are we still mutating `$global->{app_name}` before this assertion?

Reply:
Ah, no, I must have dropped that by mistake. I'll re-instate that. Good catch!